PWGCF/Flow/Core/FlowJSPCAnalysis.cxx

```diff
@@ -0,0 +1,420 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+// Header files.
+
+// O2 headers.
+
+// O2 Physics headers.
+
+#include "PWGCF/Flow/Core/FlowJSPCAnalysis.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace std;
+
+namespace o2::analysis::PWGCF
+{
+
+void FlowJSPCAnalysis::CalculateQvectors(const vector<float>& phiAngles, const vector<float>& phiWeights)
```

Comment:
Is not this functionality already addressed in the JCORRAN package from the Jyväskylä under PWGCF?
If not, cannot be the Flow Generic Framework implementation currently under PWGCF being used instead?

Reply:
I couldn't find a method which would not assume the size of the Q-vectors, the n-harmonics and powers. As those change analysis to analysis, tt would be beneficial to define a general method, which doesn't make this assumption. I'll discuss this with Jasper as well.

Reply:
We decided that we'd create a new CalculateQVectors template that can be used by all analyses under JCorran.

Reply:
I've made a PR here https://github.com/AliceO2Group/O2Physics/pull/5632
Maxim will then make the changes around that